src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -1323,6 +1323,18 @@ static constexpr Property<std::vector<std::string>, PropertyMutability::RO> exec
  */
 static constexpr Property<std::string, PropertyMutability::RW> weights_path{"WEIGHTS_PATH"};
 
+/** @brief The blob stream type to pass as property. */
+using BlobStream = std::reference_wrapper<std::istream>;
+
+/** @brief  The stream to model blob.
+ * @ingroup ov_runtime_cpp_prop_api
+ *
+ * The property is used pass model stream to the plugin for model compilation.
+ * The stream can be regular or weightless model. The `weights_path` property will work as hint where to look for
+ * weights. If compiled model fail from blob stream then original model will be used for compilation.
+ */
+inline constexpr Property<BlobStream, PropertyMutability::RW> blob_stream{"BLOB_STREAM"};
```

Comment:
one of the concerns from NPU team was that stream requires data copy when `blob_stream` is used for blob with weights.

Even as WA for stream issue, we have https://github.com/openvinotoolkit/openvino/pull/27981, it's assumed that plugins can partially parse `ov::internal::cached_model_buffer` or set ROI for places with weights.
BTW, looks like `ov::internal::cached_model_buffer` is the same property as you adding, but with different underlying data type.

Maybe we need to align such internal and public properties to have the same data type?
With @PatrikStepan we discussed that we can use `ov::Tensor` which can be mmap-ed via new `read_tensor_data` function introduced in https://github.com/openvinotoolkit/openvino/pull/29175

Reply:
To continue: we can create a property which holds `ov::Tensor` and can be created via both:
- ov::Tensor - when users decrypted compiled_blob on app side and then passes ov::Tensor
- file_path - when blob is not encrypted initially and users want OpenVINO to read blob. Such ctor can read ov::Tensor via mmap internally.

So, to plugins such property always provide ov::Tensor and plugins don't need to handle path / tensor cases manually.

Reply:
Yes, working on it. It will be simpler if user provide model blob as tensor then plugin can handle .
Should I re-name this property name to `COMPILED_BLOB` ?

Reply:
Initially, property are supposed  to be named as `ov::hint::compiled_blob`.

"Hint" denotes that such property can contain a compiled blob that cannot be compatible with current runtime, etc and exception should not be thrown.

Reply:
The `ov::hint::compiled_blob` property has been added and used instead `cached_model_buffer`